packages/broker/test/unit/ConfigWizard.test.ts

```diff
@@ -1,178 +1,202 @@
+            expect(validate(privateKey)).toBe(`Invalid privateKey provided for import: ${privateKey}`)
+        })
     })
 
-    afterEach(async () => {
-        // @ts-expect-error
-        wizard.inquirer.prompt = () => Promise.resolve()
-    })
+    describe('plugin port validation', () => {
+        it ('happy path', () => {
+            const validate = validatePortPrompt.validate!
+            expect(validate('7070')).toBe(true)
+        })
 
-    it('should generate a valid privateKey', async () => {
-        mockPromptMethod(wizard, {
-            generateOrImportEthereumPrivateKey: 'generate'
-        })
-        const wallet: Wallet = await wizard.generateOrImportPrivateKey()
-        expect(wallet.privateKey.length).toBe(66)
-        expect(wallet.privateKey.charAt(0)).toBe('0')
-        expect(wallet.privateKey.charAt(1)).toBe('x')
-        expect(wizard.config.ethereumPrivateKey).toBe(wallet.privateKey)
-        expect(wallet.address.length).toBe(42)
-        expect(wallet.address.charAt(0)).toBe('0')
-        expect(wallet.address.charAt(1)).toBe('x')
+        it ('invalid data: out-of-range number', () => {
+            const validate = validatePortPrompt.validate!
+            const port = '10000000000'
+            expect(validate(port)).toBe(`Out of range port ${port} provided (valid range 1024-49151)`)
+        })
+
+        it ('invalid data: non-numeric', () => {
+            const validate = validatePortPrompt.validate!
+            const port = 'Not A Number!'
+            expect(validate(port)).toBe(`Non-numeric value provided`)
+        })
     })
 
-    it ('should import a valid privateKey', async() => {
-        const privateKey = '0x8223c67c3c430b75931a4a1620434208749120f7d37204e93bf877e5014499ed'
-        mockPromptMethod(wizard, {
-            generateOrImportEthereumPrivateKey: 'import',
-            privateKey
+    describe('storagePrompt validation', () => {
+        let tmpDataDir: string
+
+        beforeAll(() => {
+            tmpDataDir = mkdtempSync(path.join(os.tmpdir(), 'broker-test-config-wizard'))
+        })
+
+        it ('happy path', () => {
+            const validate = selectDestinationPathPrompt.validate!
+            const validPath = tmpDataDir + '/test-config.json'
+            expect(validate(validPath)).toBe(true)
+        })
+
+        it ('invalid path provided', () => {
+            const validate = selectDestinationPathPrompt.validate!
+            const invalidPath = 'invalid-path'
+            const answers: any = {}
+            const isValid = validate(invalidPath, answers)
+            expect(isValid).toBe(false)
+            expect(answers.parentDirExists).toBe(false)
+            expect(answers.fileExists).toBe(false)
+
         })
 
-        const importedWallet: Wallet = await wizard.generateOrImportPrivateKey()
-        expect(importedWallet.privateKey).toEqual(privateKey)
+        it ('happy path with overwrite destination', () => {
```

Comment:
The tested functionality was bit unclear (if the similar test at line 101 does test the overwrite functionality then maybe this could be removed)